DISCUSS: Don't modify the root user on table create

ddanielr opened the pull request:
There is no need to add additional permissions to the root user since that user already has system privileges.

Confirms this works as expected with uno. Need to add tests.

ddanielr committed:
Reduce unnecessary permissions modification

When a table is created by a user that has the SYSTEM permission there
is no need to add additional per-table permissions to that user

Likewise, if the user has the SYSTEM permission, then table actions should be allowed.

dlmarion commented:
I thought that a user could make it such that root does not have permissions on objects. For example:

1. Root user creates user1
2. Root user gives user1 permission to create namespace
3. user1 creates a namespace and some tables in it

I don't think at this point root can scan the tables in that namespace or do anything with them unless user1 gives root those permissions. Am I mis-remembering?

ddanielr committed:
Adds root permissions test

Adds a test to verify that the root user is unable to scan a user
created table in a namespace where the root user has SYSTEM level
permissions.

ddanielr committed:
update test to pass

keith-turner commented:
Would being able to run command as another user in the shell be help for this situation?  

```
root> as user1 createtable table1
root> as user2 createtable table2
```

ctubbsii commented:
The intended behavior is that the root user (or any user granted "SYSTEM" permission does not automatically get additional privileges to read/write/alter user tables). This follows the principle of least privilege, where the root user (or a delegated user with "SYSTEM" permission) has permissions to administer the system, but does not automatically have permission to do things to user data. As a privileged user, the root user (or delegate) can grant themselves permissions, but that is intentionally a second step. This limits the harm a privileged user can do unintentionally to user data.

This follows conventions in traditional RDBMS systems where the root/system user does not automatically have read/write/alter table permissions for a user's table in a database.
